Dockerfile.windows

```diff
@@ -1,57 +1,96 @@
+# Path
+RUN setx /M Path "c:\ProgramData\chocolatey\bin;c:\Program Files\Git\usr\bin;%Path%;c:\gcc\bin;c:\7zip"
 
-RUN powershell -command \
-    .\go.msi /quiet ; \
-    Expand-Archive gcc.zip \gcc -Force ; \
-    Expand-Archive runtime.zip \gcc -Force ; \
-    Expand-Archive binutils.zip \gcc -Force
+# Download everything else we need to install
+# We want a 64-bit make.exe, not 16 or 32-bit. This was hard to find, so documenting the links
+#  - http://sourceforge.net/p/mingw-w64/wiki2/Make/ -->
+#  - http://sourceforge.net/projects/mingw-w64/files/External%20binary%20packages%20%28Win64%20hosted%29/ -->
+#  - http://sourceforge.net/projects/mingw-w64/files/External binary packages %28Win64 hosted%29/make/
+RUN powershell -command sleep 5; curl.exe -L -k -o make.zip http://downloads.sourceforge.net/project/mingw-w64/External%20binary%20packages%20%28Win64%20hosted%29/make/make-3.82.90-20111115.zip
+RUN powershell -command sleep 5; curl.exe -L -k -o gcc.zip http://downloads.sourceforge.net/project/tdm-gcc/TDM-GCC%205%20series/5.1.0-tdm64-1/gcc-5.1.0-tdm64-1-core.zip 
+RUN powershell -command sleep 5; curl.exe -L -k -o runtime.zip http://downloads.sourceforge.net/project/tdm-gcc/MinGW-w64%20runtime/GCC%205%20series/mingw64runtime-v4-git20150618-gcc5-tdm64-1.zip
+RUN powershell -command sleep 5; curl.exe -L -k -o binutils.zip http://downloads.sourceforge.net/project/tdm-gcc/GNU%20binutils/binutils-2.25-tdm64-1.zip
+RUN powershell -command sleep 5; curl.exe -L -k -o 7zsetup.exe http://www.7-zip.org/a/7z1514-x64.exe 
+RUN powershell -command sleep 5; curl.exe -L -k -o lzma.7z http://www.7-zip.org/a/lzma1514.7z
+RUN powershell -command sleep 5; curl.exe -L -k -o go.msi https://storage.googleapis.com/golang/go%GOLANG_VERSION%.windows-amd64.msi
 
-RUN del go.msi gcc.zip runtime.zip binutils.zip
+# Install and expand the bits we downloaded. 
+# Note: The 7z and go.msi installers execute asynchronously. 
+RUN powershell -command .\7zsetup /S /D="c:/7zip" ; sleep 5
+RUN powershell -command .\go.msi /quiet ; wait-process msiexec
+RUN powershell -command Expand-Archive gcc.zip \gcc -Force 
+RUN powershell -command Expand-Archive runtime.zip \gcc -Force 
+RUN powershell -command Expand-Archive binutils.zip \gcc -Force
+RUN powershell -command 7z e lzma.7z bin/lzma.exe
+RUN powershell -command 7z x make.zip  make-3.82.90-20111115/bin_amd64/make.exe 
+RUN powershell -command mv make-3.82.90-20111115/bin_amd64/make.exe /gcc/bin/ 
 
-RUN powershell -Command \
-    sleep 2 ; \
-    git clone https://github.com/akavel/rsrc.git c:\go\src\github.com\akavel\rsrc ; \
-    cd \go\src\github.com\akavel\rsrc ; \
-    git checkout -q $env:RSRC_COMMIT ; \
-    go install -v
-
-WORKDIR c:/go/src/github.com/docker/docker
+# RSRC for manifest and icon	
+RUN powershell -command sleep 5 ; git clone https://github.com/akavel/rsrc.git c:\go\src\github.com\akavel\rsrc 
+RUN cd c:/go/src/github.com/akavel/rsrc && git checkout -q %RSRC_COMMIT% && go install -v
```

Comment:
I don't know if there's a convention to use git for stuff like this, but if you don't want the git dependency, you can also just curl or wget `https://github.com/akavel/rsrc/archive/ba14da1f827188454a4591717fff29999010887f.zip` (with `-L` to follow redirect).